### PR TITLE
max and min methods for NonEmptyList

### PR DESCRIPTION
**Pull request by dorians**
solves [#2593](https://github.com/arrow-kt/arrow/issues/2593)

### CODE REVIEW

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptyList.kt [hunk 1]
```diff
@@ -383,6 +383,12 @@ public class NonEmptyList<out A>(
       map(head, b.head, c.head, d.head, e.head, f.head, g.head, h.head, i.head, j.head),
       tail.zip(b.tail, c.tail, d.tail, e.tail, f.tail, g.tail, h.tail, i.tail, j.tail, map)
     )
+
+  public inline fun <B : Comparable<B>> minBy(selector: (A) -> B): A =
+    minByOrNull(selector) ?: head
```
Comment (hoc081098) [inline arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptyList.kt:388]: Dont need `?: head`, just `minByOrNull(selector)!!`

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptyList.kt [hunk 2]
```diff
@@ -383,6 +383,12 @@ public class NonEmptyList<out A>(
...
+
+  public inline fun <B : Comparable<B>> maxBy(selector: (A) -> B): A =
+    maxByOrNull(selector) ?: head
```
Comment (hoc081098) [inline arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptyList.kt:391]: Same as above

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptyList.kt [hunk 3]
```diff
@@ -397,6 +403,12 @@ public operator fun <A : Comparable<A>> NonEmptyList<A>.compareTo(other: NonEmpt
 public fun <A> NonEmptyList<NonEmptyList<A>>.flatten(): NonEmptyList<A> =
   this.flatMap(::identity)
 
+public inline fun <T : Comparable<T>> NonEmptyList<T>.min(): T =
+  minOrNull() ?: head
```
Comment (hoc081098) [inline arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptyList.kt:407]: Same as above

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptyList.kt [hunk 4]
```diff
@@ -397,6 +403,12 @@ public operator fun <A : Comparable<A>> NonEmptyList<A>.compareTo(other: NonEmpt
...
+
+public inline fun <T : Comparable<T>> NonEmptyList<T>.max(): T =
+  maxOrNull() ?: head
```
Comment (hoc081098) [inline arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptyList.kt:410]: Same as above

File: arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptyList.kt (hunk 1 above)
Comment (nomisRev) [reply arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptyList.kt:388]: `!!` should indeed be safe here, I think it results in a faster binary but not 100% sure actually.

**Review by nomisRev — COMMENTED** [review]
@dorians if you merge latest `main` it should fix the failure you see on CI.
We had some issues with CI being flaky.

**Review by nomisRev — APPROVED** [review]

**Comment by nomisRev** [comment]
@dorians can you run `./gradlew apiDump` and check in the files?
Not sure why the Github Action to do so is not working 🤔 

```
[main d[39](https://github.com/arrow-kt/arrow/runs/5084839593?check_suite_focus=true#step:6:39)7bfbd4] Update API files
 Author: raulraja <raulraja@users.noreply.github.com>
 1 file changed, 1 insertion(+)
INPUT_TAGGING_MESSAGE: 
No tagging message supplied. No tag will be added.
INPUT_PUSH_OPTIONS: 
remote: Permission to arrow-kt/arrow.git denied to github-actions[bot].
fatal: unable to access 'arrow-kt/arrow': The requested URL returned error: [40](https://github.com/arrow-kt/arrow/runs/5084839593?check_suite_focus=true#step:6:40)3
Error: Invalid status code: 128
    at ChildProcess.<anonymous> (/home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4.13.1/index.js:17:19)
    at ChildProcess.emit (events.js:210:5)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5) {
  code: 128
}
Error: Invalid status code: 128
    at ChildProcess.<anonymous> (/home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4.13.1/index.js:17:19)
    at ChildProcess.emit (events.js:210:5)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
```

**Review by raulraja — APPROVED** [review]
thanks @dorians !

**Review by i-walker — APPROVED** [review]

**Comment by dorians** [comment]
yes, I still want that to be merged :)

**Pull request merged**